src/ring_theory/algebra_tower.lean

```diff
@@ -200,6 +200,12 @@ begin
   { simp [hi] },
 end
 
+lemma basis.algebra_map_injective {ι : Type v₁} [no_zero_divisors R] [nontrivial S]
```

Comment:
I didn't find an obvious file for this result, and I didn't feel like this is worth creating a whole new file for. So I ended up putting it here since it is the meet of `algebra/algebra/basic.lean` and `linear_algebra/basis.lean`.